internal/controller/controller.go

```diff
@@ -195,22 +221,30 @@ func (c *CaddyController) handleErr(err error, action interface{}) {
 	logrus.Error(err)
 }
 
-// loadConfigFromFile loads caddy with a config defined by an io.Reader.
-func (c *CaddyController) loadConfigFromFile(cfg io.Reader) error {
-	buf := new(bytes.Buffer)
-	buf.ReadFrom(cfg)
+func loadCaddyConfigFile(cfgPath string) (*c.Config, error) {
+	var caddyCfgMap *c.Config
+	if _, err := os.Stat(cfgPath); !os.IsNotExist(err) {
+		file, err := os.Open(cfgPath)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+
+		b, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, err
+		}
 
-	err := caddy.Load(buf.Bytes(), true)
-	if err != nil {
-		return fmt.Errorf("could not load caddy config %v", err.Error())
+		json.Unmarshal(b, &caddyCfgMap)
+	} else {
+		return nil, nil
```

Comment:
I have some doubt on generating a nil Config without error 🤔

Reply:
I'd say it makes sense here because there is no error per se. I don't see another way to make config file optional, any ideas (I'm still not very familiar with golang 😄)?